include/bluetooth/gatt.h

```diff
@@ -100,44 +100,55 @@ enum {
 	BT_GATT_WRITE_FLAG_CMD = BIT(1),
 };
 
+/* Forward declaration of GATT Attribute structure */
+struct bt_gatt_attr;
+
+/** @typedef bt_gatt_attr_read_func_t
+ *  @brief Attribute read callback
+ *
+ *  The callback can also be used locally to read the contents of the
+ *  attribute in which case no connection will be set.
+ *
+ *  @param conn   The connection that is requesting to read
+ *  @param attr   The attribute that's being read
+ *  @param buf    Buffer to place the read result in
+ *  @param len    Length of data to read
+ *  @param offset Offset to start reading from
+ *
+ *  @return Number of bytes read, or in case of an error
+ *          BT_GATT_ERR() with a specific ATT error code.
+ */
+typedef ssize_t (*bt_gatt_attr_read_func_t)(struct bt_conn *conn,
+					    const struct bt_gatt_attr *attr,
+					    void *buf, uint16_t len,
+					    uint16_t offset);
```

Comment:
I like the changes you did for the enum and the macros, but I actually prefer defining callbacks as functions inside the scope of the callback structure, but we'll see if others agree/disagrees with this.

Reply:
Thanks for the review, @Thalley! Yeah, I see how defining callbacks in place can be more compact. However, that way it's harder to refer to them from other places in the documentation. Let's see what others have to say.

BTW, while I (hopefully) have your ear, I'm struggling regenerating the Doxygen HTML locally after changing the header. I get `Doxygen build will be skipped (no changes)!` when running `make` in the `doc` directory (and I made sure there were changes). Even if I edit the code detecting the changes and doxygen runs, the results apparently are not copied to the output directory anyway. What am I doing wrong? Are there any docs on this I should read?

Resolving this could help me fix the issues CI has found.
Thanks!

Reply:
@spbnick Agreed, it is harder to refer to a field in a struct, but you can do something like `@ref <name_of_struct>::<name_of_field>` 
So instead of `Descriptor attribute write callback (bt_gatt_attr_write_func_t)` you could do `@ref bt_gatt_attr::write` I think. 

I haven't built documentation locally myself for a very long time, so not sure if I can help you with that issue

Reply:
> @spbnick Agreed, it is harder to refer to a field in a struct, but you can do something like @ref <name_of_struct>::<name_of_field>
Ah, I thought there should be something like that. I always forget the Doxygen refernce syntax. Yeah, I could do it like that, if there's preference. There's one more benefit for having the type defined separately, though: you can then define your own variables, e.g. for your own structures or functions generating those attributes.

Reply:
I think you have to use the dot separator in Doxygen i.e. `@ref type_name.field_name`.